### PR TITLE
[CIAS30-3660] fix "Duplicate Here" option for invited collaborator making copy to a intervention owner

### PR DESCRIPTION
**Pull request by OskarDybowskiHTD**
## Related tasks
- [CIAS-3660](https://htdevelopers.atlassian.net/browse/CIAS30-3660)

## What's new?
- Using "Duplicate here" will now clone the intervention and make the user who requested the clone the owner, instead of the owner being always the owner of the original one

### CODE REVIEW

**Review by WiktoriaRozanska — COMMENTED** [review]
Add description to this task

File: app/models/concerns/clone.rb [hunk 1]
```diff
@@ -6,10 +6,14 @@ module Clone
...
   def clone(params: {}, clean_formulas: true, position: nil, hidden: false)
     emails = params[:emails]&.map(&:downcase)
+    current_user_id = params[:user_id]
 
     if emails.blank?
-      return "Clone::#{de_constantize_modulize_name.classify}".
-        safe_constantize.new(self, clean_formulas: clean_formulas, position: position, params: params, hidden: hidden).execute
+      if current_user_id.present?
+        return clone_module.new(self, user_id: current_user_id, clean_formulas: clean_formulas, position: position, params: params, hidden: hidden).execute
+      end
```
Comment (WiktoriaRozanska) [inline app/models/concerns/clone.rb:14]: Additional "if" in this place will be confusing here. Can we move this logic to `Clone::Base` or another place? 🤔

File: app/models/concerns/clone.rb [hunk 2]
```diff
@@ -20,14 +24,17 @@ def clone(params: {}, clean_formulas: true, position: nil, hidden: false)
...
     end
     cloned_elements
   end
+
+  private
+
+  # returns the corresponding clone module (e.g. Intervention -> Clone::Intervention)
+  def clone_module
+    "Clone::#{de_constantize_modulize_name.classify}".safe_constantize
+  end
```
Comment (WiktoriaRozanska) [inline app/models/concerns/clone.rb:39]: noice 😎

File: app/models/concerns/clone.rb (hunk 1 above)
Comment (WiktoriaRozanska) [reply app/models/concerns/clone.rb]: I don't see any changes

Comment (OskarDybowski) [reply app/models/concerns/clone.rb]: Please check https://github.com/Michigan-State-University/cias-api/pull/135/commits/31e6ad15c81b8d418a8fbd83b1a7dffd6dc6eacf

**Review by WiktoriaRozanska — APPROVED** [review]

**Pull request merged**